### PR TITLE
feat: RawStream

### PR DESCRIPTION
**Pull request by schiller-manuel**
## Summary by CodeRabbit

* **New Features**
  * Added a "Raw Stream" section with demo pages (SSR + client), an interactive client test harness, text/binary hint handling, and end-to-end streaming demos.
  * Introduced framed/multiplexed streaming to deliver JSON plus multiple binary/text streams together, with client/server framing and decoding support and public RawStream APIs.

* **Tests**
  * Added extensive unit and end-to-end tests covering framing, decoding, multiplexing, SSR hints, RPC streaming, navigation, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/router-core/src/ssr/serializer/RawStream.ts [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+        try {
+          // Try UTF-8 decode first
+          const text = decoder.decode(value, { stream: true })
+          if (text.length > 0) {
+            stream.next(text)
+          }
+        } catch {
+          // UTF-8 decode failed, fallback to base64
+          stream.next({ $b64: uint8ArrayToBase64(value) })
+        }
```
[the remark this answers is not in the document]
Comment (schiller-manuel) [reply packages/router-core/src/ssr/serializer/RawStream.ts:230]: Finding: TextDecoder poisoning after fatal UTF-8 decode error
Location: packages/router-core/src/ssr/serializer/RawStream.ts lines 209-218
Recommendation: Reset the TextDecoder after catching a decode error to prevent subsequent chunks from being tainted by the decoder's internal buffer.
---
Investigation Results:
I investigated this concern by testing the actual behavior of TextDecoder with fatal: true in Node.js. The hypothesis was that after a fatal decode error, the decoder's internal buffer would retain partial multi-byte sequences that would corrupt subsequent decodes.
Test scenarios executed:
1. Valid → Invalid → Valid UTF-8 chunks
2. Partial multi-byte sequence → Error → New valid sequence
3. Incomplete emoji (4-byte) → Invalid continuation → Complete new sequence
Findings:
In all test cases, the Node.js TextDecoder implementation automatically clears its internal buffer after a fatal error. Subsequent valid UTF-8 chunks decode correctly without needing to recreate the decoder.
Example test:
const decoder = new TextDecoder('utf-8', { fatal: true });
decoder.decode(new Uint8Array([0xF0, 0x9F]), { stream: true }); // partial emoji
try { decoder.decode(new Uint8Array([0x41]), { stream: true }); } catch(e) {} // error
const result = decoder.decode(new Uint8Array([0xC4, 0x80]), { stream: true }); // Ā
console.log(result); // Output: "Ā" (correctly decoded, buffer was cleared)
Conclusion:
The WHATWG TextDecoder spec behavior in modern runtimes (Node.js, browsers) clears internal state after fatal errors. The suggested fix would add unnecessary overhead (creating a new TextDecoder on every error) without providing any practical benefit.
Recommendation: No change needed. The current implementation is correct for all tested runtimes.
